SignInCheckIn/SignInCheckIn/Services/RoomService.cs

```diff
@@ -35,5 +45,68 @@ public EventRoomDto CreateOrUpdateEventRoom(string authenticationToken, EventRoo
 
             return Mapper.Map<EventRoomDto>(response);
         }
+
+        public List<AgeGradeDto> GetEventRoomAgesAndGrades(string authenticationToken, int eventId, int roomId)
```

Comment:
@timmyg is there any way to break this method up.  The logic looks good but it takes a long time to break it down to understand what each part is doing.